runtime/lua/vim/health.lua

```diff
@@ -331,13 +331,24 @@ function M._check(mods, plugin_names)
 
   local emptybuf = vim.fn.bufnr('$') == 1 and vim.fn.getline(1) == '' and 1 == vim.fn.line('$')
 
-  -- When no command modifiers are used:
-  -- - If the current buffer is empty, open healthcheck directly.
-  -- - If not specified otherwise open healthcheck in a tab.
-  local buf_cmd = #mods > 0 and (mods .. ' sbuffer') or emptybuf and 'buffer' or 'tab sbuffer'
-
   local bufnr = vim.api.nvim_create_buf(true, true)
-  vim.cmd(buf_cmd .. ' ' .. bufnr)
+  if #mods > 0 and mods == 'float' then
```

Comment:
can we re-use the lsp floating window util? maybe extract it to a private `vim._util` module for now.

Reply:
`vim.lsp.open_floating_preview` take content and create float seems like not fit here

Reply:
`contents` is just used to guess the window size. we can just pass the healthcheck contents, what's the problem?